lib/CL/pocl_llvm_api.cc

```diff
@@ -1146,7 +1161,12 @@ static PassManager& kernel_compiler_passes
   }
   passes.push_back("workgroup");
   passes.push_back("allocastoentry");
+#ifdef HSA_RUNTIME_IS_ROCM
```

Comment:
Ditto. SPMD does not imply a relation to TAS. This should be cleaned up. A device hook or similar.